dns/dns_updater.go

```diff
@@ -167,18 +103,39 @@ func (u *updater) Update(entries controller.IngressEntries) error {
 	return nil
 }
 
-func (u *updater) determineManagedRecordSets(rrs []*route53.ResourceRecordSet) []*route53.ResourceRecordSet {
+func (u *updater) consolidateRecordsFromRoute53(rrs []*route53.ResourceRecordSet) []consolidatedRecord {
+	var records []consolidatedRecord
+
+	for _, recordSet := range rrs {
+		if *recordSet.Type == route53.RRTypeA && recordSet.AliasTarget != nil {
+			records = append(records, consolidatedRecord{
+				name:            *recordSet.Name,
+				pointsTo:        *recordSet.AliasTarget.DNSName,
+				aliasHostedZone: *recordSet.AliasTarget.HostedZoneId,
+			})
+		} else if *recordSet.Type == route53.RRTypeCname {
+			records = append(records, consolidatedRecord{
+				name:     *recordSet.Name,
+				pointsTo: *recordSet.ResourceRecords[0].Value,
+			})
+		}
+	}
+
+	return records
+}
+
+func (u *updater) determineManagedRecordSets(rrs []consolidatedRecord) []consolidatedRecord {
 	managedLBs := make(map[string]bool)
 	for _, dns := range u.schemeToDNS {
 		managedLBs[dns.dnsName] = true
 	}
-	var managed []*route53.ResourceRecordSet
+	var managed []consolidatedRecord
 	var nonManaged []string
 	for _, rec := range rrs {
-		if rec.AliasTarget != nil && rec.AliasTarget.DNSName != nil && managedLBs[*rec.AliasTarget.DNSName] {
+		if rec.name != "" && managedLBs[rec.pointsTo] {
```

Comment:
records with no dns name used to be excluded `rec.AliasTarget.DNSName != nil`, but I saw no test for that, so I'm unsure whether this is still needed

Reply:
CNAME records won't have an alias target, so that check isn't appropriate any more.

Reply:
this check was done on A records.
However looking at the ResourceRecordSet.AliasTarget doc, I can see that DNSName is a required field, so that check was redundant